app/services/record_declarations/base.rb

```diff
@@ -106,9 +96,56 @@ def create_record!
       end
     end
 
+    def valid_courses_for_user
+      valid_courses = []
+      valid_courses << "ecf-mentor" if user.mentor?
+      valid_courses << "ecf-induction" if user.early_career_teacher?
+      valid_courses += NPQCourse.all.map(&:identifier) if user.npq?
+      valid_courses
+    end
+
+    def profile_exists
+      return if errors.any?
+
+      unless user_profile
+        errors.add(:user_profile, I18n.t(:invalid_participant))
+      end
+    end
+
+    def date_has_the_right_format
+      return if declaration_date.blank?
+
+      errors.add(:declaration_date, I18n.t(:invalid_declaration_date)) unless declaration_date.match(RFC3339_DATE_REGEX)
+      errors.add(:declaration_date, I18n.t(:future_declaration_date)) if parsed_date > Time.zone.now
```

Comment:
I'd have expected to see the format and business future date logic as separate validations.